frontend/javascripts/oxalis/controller/combinations/tool_controls.js

```diff
@@ -0,0 +1,271 @@
+import PlaneView from "oxalis/view/plane_view";
+import Store from "oxalis/store";
+import {
+  type OrthoView,
+  OrthoViews,
+  AnnotationToolEnum,
+  type Point2,
+  ContourModeEnum,
+  type ShowContextMenuFunction,
+} from "oxalis/constants";
+import { type ModifierKeys } from "libs/input";
+import * as Utils from "libs/utils";
+import api from "oxalis/api/internal_api";
+import * as SkeletonHandlers from "oxalis/controller/combinations/skeleton_handlers";
+import * as VolumeHandlers from "oxalis/controller/combinations/volume_handlers";
+import * as MoveHandlers from "oxalis/controller/combinations/move_handlers";
+import { handleAgglomerateSkeletonAtClick } from "oxalis/controller/combinations/segmentation_handlers";
+import {
+  getContourTracingMode,
+  enforceVolumeTracing,
+} from "oxalis/model/accessors/volumetracing_accessor";
+
+/*
+  This module contains classes for the different tools, such as MoveTool, SkeletonTool, DrawTool etc.
+  Each tool class defines getMouseControls which declares how mouse bindings are mapped (depending on
+  modifiers) to actions. For the actions, code from oxalis/controller/combinations is called.
+
+  If a tool does not define a specific mouse binding, the bindings of the MoveTool are used as a fallback.
+  See `createToolDependentMouseHandler` in plane_controller.js
+*/
+
+export class MoveTool {
+  static getMouseControls(
+    planeId: OrthoView,
+    planeView: PlaneView,
+    showNodeContextMenuAt: ShowContextMenuFunction,
+  ): Object {
+    return {
+      scroll: (delta: number, type: ?ModifierKeys) => {
+        switch (type) {
+          case null: {
+            MoveHandlers.moveZ(delta, true);
+            break;
+          }
+          case "alt":
+          case "ctrl": {
+            MoveHandlers.zoomPlanes(Utils.clamp(-1, delta, 1), true);
+            break;
+          }
+          case "shift": {
+            const { uiInformation, tracing } = Store.getState();
+            const isBrushActive = uiInformation.activeTool === AnnotationToolEnum.BRUSH;
+            if (isBrushActive) {
+              // Different browsers send different deltas, this way the behavior is comparable
+              if (delta > 0) {
+                VolumeHandlers.changeBrushSizeIfBrushIsActiveBy(1);
+              } else {
+                VolumeHandlers.changeBrushSizeIfBrushIsActiveBy(-1);
+              }
+            } else if (tracing.skeleton) {
+              // Different browsers send different deltas, this way the behavior is comparable
+              api.tracing.setNodeRadius(delta > 0 ? 5 : -5);
+            }
+            break;
+          }
+          default: // ignore other cases
+        }
+      },
+      over: () => {
+        MoveHandlers.handleOverViewport(planeId);
+      },
+      pinch: delta => MoveHandlers.zoom(delta, true),
+      mouseMove: (delta: Point2, position: Point2, id, event) => {
+        // Always set the correct mouse position. Otherwise, using alt + mouse move and
```

Comment:
As this will result in many store updates, maybe we should check that there is no component that is rendered on every store update. Using the React Dev Tools with the "Highlight Updates" setting should show that.
However, it's probably not that important as this used to be done already in volume tracings to update the displayed segment id.

Reply:
Good point! I think the store update itself is not that bad since (as you say) it's already happening for volume tracings (which is the default due to hybrids). However, the "highlight updates" hint is still very relevant. I noticed that the toolbar updates now on each mouse move. I added this to-do element to #5422 and will tackle this after the other functional changes have been finished.